### ISSUE TITLE
[Popup] If a fragment anchor is located inside a popup, should it open upon navigation?

### ISSUE THREAD

**Issue by mfreed7** [WICG/open-ui#315]
For example, here:

```html
<popup id=foo>
   <div id=bar>Hello</div>
</popup>
```

Should navigating to the `#foo` or `#bar` fragments automatically open the popup?

With that behavior, some of the use cases listed in https://github.com/WICG/open-ui/issues/311 could be declaratively achieved, without needing an `open` attribute.

Some corner cases that might be interesting:

```
<popup id=p></popup>
<button popup=p id=anchor></button>
```
==> should navigating to `#anchor` open the popup?

```
<popup>
  <popup>
    <span id=anchor></span>
  </popup>
</popup>
```
==> should only the innermost popup be shown? The issue here is that it won't be visible - it is nested inside another non-shown `<popup>`, which is `display:none`.

I think the popup stack behavior is pretty clear in this case: navigating to a fragment anchor like this should behave as if `.show()` was called on the relevant popup. So if there's already another popup showing when the navigation happens, the existing popup is closed.

@flackr @mirisuzanne @stubbornella

**Comment by mirisuzanne** [comment]
This use-case came up in a conversation about popups, and I think that's useful. I can imagine linking to a specific menu-item, for example. But I think the pattern becomes even more useful when applied to more persistent elements like a `dialog` or `details`.

**Comment by mfreed7** [comment]
Another similar issue, which might deserve its own issue, was brought up at today's OpenUI meeting: should Find in Page be able to discover the content of a hidden `<popup>`? And should Find in Page be able to **show** the popup when that entry is selected by Find in Page?

Similar issues are being tackled for `<summary>`/`<details>`, via content visibility, so perhaps there's some overlap here. @josepharhar @chrishtr @tantek

**Comment by hidde** [comment]
I agree with @mirisuzanne and think it is useful for popups, `dialog` and `details`/`summary`. 

On the WAI Website we have a script that checks if the URL has a fragment identifier and opens the relevant `details` element, as we found people would want to see that opened if they link to it ([example: fragment link where anchor is inside `details`](https://www.w3.org/WAI/teach-advocate/accessible-presentations/#benefits-organizers--speakers)).
